Lib/test/test_concurrent_futures.py

```diff
@@ -568,6 +568,19 @@ def test_shutdown_no_wait(self):
         # shutdown.
         assert all([r == abs(v) for r, v in zip(res, range(-5, 5))])
 
+    def test_hang_issue45021(self):
+        """https://bugs.python.org/issue45021"""
+        def submit(pool):
+            pool.submit(submit, pool)
+
+        if hasattr(os, 'register_at_fork'):
```

Comment:
You should decorate the test method with: ``@unittest.skipUnless(hasattr(os, 'register_at_fork'), 'need os.register_at_fork')``

Reply:
I also moved the test in ThreadPoolExecutorTests where it gets executed only once